scale-encode/src/error/mod.rs

```diff
@@ -83,64 +83,73 @@ impl Error {
 }
 
 impl Display for Error {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
         let path = self.context.path();
         let kind = &self.kind;
         write!(f, "Error at {path}: {kind}")
     }
 }
 
 /// The underlying nature of the error.
-#[derive(Debug, thiserror::Error)]
+#[derive(Debug)]
 pub enum ErrorKind {
     /// Cannot find a given type.
-    #[error("Cannot find type with ID {0}")]
     TypeNotFound(u32),
     /// Cannot encode the actual type given into the target type ID.
-    #[error("Cannot encode {actual:?} into type with ID {expected}")]
     WrongShape {
         /// The actual kind we have to encode
         actual: Kind,
         /// ID of the expected type.
         expected: u32,
     },
     /// The types line up, but the expected length of the target type is different from the length of the input value.
-    #[error("Cannot encode to type; expected length {expected_len} but got length {actual_len}")]
     WrongLength {
         /// Length we have
         actual_len: usize,
         /// Length expected for type.
         expected_len: usize,
     },
     /// We cannot encode the number given into the target type; it's out of range.
-    #[error("Number {value} is out of range for target type {expected}")]
     NumberOutOfRange {
         /// A string represenatation of the numeric value that was out of range.
         value: String,
         /// Id of the expected numeric type that we tried to encode it to.
         expected: u32,
     },
     /// Cannot find a variant with a matching name on the target type.
-    #[error("Variant {name} does not exist on type with ID {expected}")]
     CannotFindVariant {
         /// Variant name we can't find in the expected type.
         name: String,
         /// ID of the expected type.
         expected: u32,
     },
     /// Cannot find a field on our source type that's needed for the target type.
-    #[error("Field {name} does not exist in our source struct")]
     CannotFindField {
         /// Name of the field which was not provided.
         name: String,
     },
     /// A custom error.
-    #[error("Custom error: {0}")]
     Custom(CustomError),
 }
 
+impl From<CustomError> for ErrorKind {
+    fn from(err: CustomError) -> ErrorKind {
+        ErrorKind::Custom(err)
+    }
+}
+
+impl Display for ErrorKind {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(f, "{self:?}")
```

Comment:
Small thing: we lose the nice error texts here and just debug log the thing instead. I think it'd be better to do something like:

```
match self {
    ErrorKind::TypeNotFound(id) => write!(f, "Cannot find type with ID {id}"),
    // .. and so on
}
```

Just so that the Display impl is a little nicer than the Debug one and as it was

Reply:
Agreed, that's the more elegant way to solve this